wallet/wallet.h

```diff
@@ -313,8 +313,9 @@ static inline enum channel_state channel_state_in_db(enum channel_state s)
 		BUILD_ASSERT(CHANNELD_AWAITING_SPLICE == 13);
 		return s;
 	case DUALOPEND_OPEN_INIT:
-		/* Never appears in db! */
-		break;
+		/* We save channels that aren't yet committed :( */
+		BUILD_ASSERT(DUALOPEND_OPEN_INIT == 1);
+		return s;
 	}
```

Comment:
Absolutely not.  Not again.

You need a new state, DUALOPEND_OPEN_COMMITTED.  When you add it, the compiler will tell you everywhere you have to change it.